js/rmdir.ts

```diff
@@ -0,0 +1,37 @@
+// Copyright 2018 the Deno authors. All rights reserved. MIT license.
+import * as fbs from "gen/msg_generated";
+import { flatbuffers } from "flatbuffers";
+import * as dispatch from "./dispatch";
+
+/**
+ * Removes a directory with the specified path synchronously.
+ *
+ *     import { rmdirSync } from "deno";
+ *     rmdirSync("dir");
+ */
+export function rmdirSync(path: string, recursive = false): void {
+  dispatch.sendSync(...req(path, recursive));
+}
+
+/**
+ * Removes a directory with the specified path.
+ *
+ *     import { rmdir } from "deno";
+ *     await rmdir("dir");
+ */
+export async function rmdir(path: string, recursive = false): Promise<void> {
+  await dispatch.sendAsync(...req(path, recursive));
+}
```

Comment:
In terms of the public API, I think we should have separate functions for the recursive function. Internally you can use a flag to reduce code.

Please change the API:
- remove() and removeSync()   (currently called rmdir and rmdirSync)
- removeAll() and removeAllSync (same but with recursive flag)

And please this file to js/remove.ts

Reply:
Sure, feels more readable. Although I do feel probably something like `removeDir` would be clearer? (to avoid confusing with removing a file)

Currently at office. Would finish working on this by evening.

Reply:
In libc "remove" handles both directories and files: [remove(3)](https://linux.die.net/man/3/remove)

Reply:
Got it. Will combine the 2 usages into a single function.